build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/TestingConventionsPrecommitPlugin.java

```diff
@@ -8,41 +8,79 @@
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.api.tasks.SourceSet;
 import org.gradle.api.tasks.TaskProvider;
-import org.gradle.api.tasks.testing.Test;
 
-import java.util.stream.Collectors;
+import java.util.List;
+
+public class TestingConventionsPrecommitPlugin extends PrecommitPlugin {
+
+    public static final String TESTING_CONVENTIONS_TASK_NAME = "testingConventions";
 
-public class TestingConventionsPrecommitPlugin extends PrecommitPlugin implements InternalPlugin {
     @Override
     public TaskProvider<? extends Task> createTask(Project project) {
-        TaskProvider<TestingConventionsTasks> testingConventions = project.getTasks()
-            .register("testingConventions", TestingConventionsTasks.class, t -> {
-                TestingConventionRule testsRule = t.getNaming().maybeCreate("Tests");
-                testsRule.baseClass("org.apache.lucene.tests.util.LuceneTestCase");
-                TestingConventionRule itRule = t.getNaming().maybeCreate("IT");
-                itRule.baseClass("org.elasticsearch.test.ESIntegTestCase");
-                itRule.baseClass("org.elasticsearch.test.rest.ESRestTestCase");
-                t.setCandidateClassFilesProvider(
-                    project.provider(
-                        () -> project.getTasks()
-                            .withType(Test.class)
-                            .matching(Task::getEnabled)
-                            .stream()
-                            .collect(Collectors.toMap(Task::getPath, task -> task.getCandidateClassFiles().getFiles()))
-                    )
-                );
-                SourceSetContainer javaSourceSets = GradleUtils.getJavaSourceSets(project);
-                t.setSourceSets(javaSourceSets);
-                // Run only after everything is compiled
-                javaSourceSets.all(sourceSet -> t.dependsOn(sourceSet.getOutput().getClassesDirs()));
+        project.getPlugins().apply(JavaBasePlugin.class);
+        var javaPluginExtension = project.getExtensions().getByType(JavaPluginExtension.class);
+        var sourceSets = javaPluginExtension.getSourceSets();
+        var tasks = project.getTasks();
+
+        project.getPlugins().withType(JavaPlugin.class, javaPlugin -> {
+            NamedDomainObjectProvider<SourceSet> sourceSet = sourceSets.named(SourceSet.TEST_SOURCE_SET_NAME);
+            setupTaskPerSourceSet(project, sourceSet, t -> {
+                t.getSuffix().convention("Tests");
+                t.getBaseClasses().convention(List.of("org.apache.lucene.tests.util.LuceneTestCase"));
             });
-        return testingConventions;
+        });
+
+        project.getPlugins().withType(InternalYamlRestTestPlugin.class, yamlRestTestPlugin -> {
+            NamedDomainObjectProvider<SourceSet> sourceSet = sourceSets.named(InternalYamlRestTestPlugin.SOURCE_SET_NAME);
+            setupTaskPerSourceSet(project, sourceSet, t -> {
```

Comment:
At the moment we apply the same base class defaults for yaml and java rest tests. We might introduce a dedicated base class for each of them at one point.